pkg/controller/user/user.go

```diff
@@ -92,3 +92,36 @@ func (base *Controller) LoginUser(c *gin.Context) {
 	rd := utility.BuildSuccessResponse(http.StatusOK, "user login successfully", respData)
 	c.JSON(http.StatusOK, rd)
 }
+
+// update user details endpoint
+func (base *Controller) UpdateUser(c *gin.Context) {
+	var req models.UpdateUserRequestModel
+
+	// Bind the request body to the UpdateUserRequestModel
+	if err := c.ShouldBindJSON(&req); err != nil {
+		rd := utility.BuildErrorResponse(http.StatusBadRequest, "error", "Failed to parse request body", err, nil)
+		c.JSON(http.StatusBadRequest, rd)
+		return
+	}
+
+	// Validate the request data
+	if err := base.Validator.Struct(&req); err != nil {
+		rd := utility.BuildErrorResponse(http.StatusUnprocessableEntity, "error", "Validation failed", utility.ValidationResponse(err, base.Validator), nil)
+		c.JSON(http.StatusUnprocessableEntity, rd)
+		return
+	}
+
+	// Extract user ID from the URL parameter
+	userID := c.Param("userId")
+
```

Comment:
move this up, should be one of the first things you do in this function

Reply:
Noted sir, working on it.